Post Model Timezone test issue

### Issue Summary

When running server tests locally (using `grunt validate`) I continue to get an error that appears to be related to timezones:

The mocha test that fails: `Post Model Single author posts findOne can findOne, returning a dated permalink:`

The error message itself:

```
AssertionError: expected '/2015/01/01/html-ipsum/' to be '/2014/12/31/html-ipsum/'
+ expected - actual

-/2015/01/01/html-ipsum/
+/2014/12/31/html-ipsum/
```

### Steps to Reproduce

1. Clone 👻 
2. `npm install`
3. `grunt validate`
4. get error

It also occurs when running `grunt test-integration`. I tried deleting the `ghost-test.db` file in `content/data`, but that didn't affect anything.

### Technical details:

* Ghost Version: 0.9.0-beta.1 (also latest [master](https://github.com/TryGhost/Ghost/commit/451c29d06a65d3904470b6b30f2fb21ad5c162c2))
* Node Version: v0.12.2
* Browser/OS: Mac OSX El Capitan (10.11.5)
* Database: SQLite

--- a/core/test/integration/model/model_posts_spec.js
+++ b/core/test/integration/model/model_posts_spec.js
@@ -346,12 +346,7 @@ describe('Post Model', function () {
             });
 
             it('can findOne, returning a dated permalink', function (done) {
-                var firstPost = 1,
-                    today = testUtils.DataGenerator.Content.posts[0].published_at,
-                    dd = ('0' + today.getDate()).slice(-2),
-                    mm = ('0' + (today.getMonth() + 1)).slice(-2),
-                    yyyy = today.getFullYear(),
-                    postLink = '/' + yyyy + '/' + mm + '/' + dd + '/html-ipsum/';
+                var firstPost = 1;
 
                 configUtils.set({theme: {
                     permalinks: '/:year/:month/:day/:slug/'
@@ -361,7 +356,10 @@ describe('Post Model', function () {
                     .then(function (result) {
                         should.exist(result);
                         firstPost = result.toJSON();
-                        firstPost.url.should.equal(postLink);
+
+                        // published_at of post 1 is 2015-01-01 00:00:00
+                        // default blog TZ is UTC
+                        firstPost.url.should.equal('/2015/01/01/html-ipsum/');
 
                         done();
                     }).catch(done);

--- a/core/test/unit/config_spec.js
+++ b/core/test/unit/config_spec.js
@@ -413,6 +413,17 @@ describe('Config', function () {
                 config.urlPathForPost(testData).should.equal(postLink);
             });
 
+            it('permalink is /:year/:month/:day/:slug, blog timezone is Asia Tokyo', function () {
+                config.theme.timezone = 'Asia/Tokyo';
+                config.theme.permalinks = '/:year/:month/:day/:slug/';
+
+                var testData = testUtils.DataGenerator.Content.posts[2],
+                    postLink = '/2016/05/18/short-and-sweet/';
+
+                testData.published_at = new Date('2016-05-18T06:30:00.000Z');
+                config.urlPathForPost(testData).should.equal(postLink);
+            });
+
             it('post is page, no permalink usage allowed at all', function () {
                 config.theme.timezone = 'America/Los_Angeles';
                 config.theme.permalinks = '/:year/:month/:day/:slug/';

